pkg/cluster/loadbalancerprofile.go

```diff
@@ -95,9 +93,34 @@ func (m *manager) reconcileOutboundRuleV4IPsInner(ctx context.Context, lb mgmtne
 	return nil
 }
 
-// Remove all frontend ip config in use by outbound-rule-v4.  Frontend IP config that is used by load balancer rules will be saved.
+func replaceOutboundRuleV4OutboundIPs(resourceGroupID string, lb mgmtnetwork.LoadBalancer, desiredOutboundIPs []api.ResourceReference) {
```

Comment:
unless we plan on testing this function separately from `reconcileLoadBalancerProfile`, I don't see the need to split these two calls into their own func.

The same goes for `removeOutboundIPsFromLB`, since all the contained functions are acting on side effects, IMO it makes more sense to have them be called directly by the parent and not nest them two levels.

Reply:
so this would mean line 80 inside `reconcileLoadBalancerProfile` calls the now three functions in a row:
```
removeOutboundRuleV4FrontendIPConfig(lb)

setOutboundRuleV4(lb, []mgmtnetwork.SubResource{})

addOutboundIPsToLB(resourceGroupID, lb, desiredOutboundIPs)
```

Reply:
I split this up to maybe make it more readable, but I can agree on that its not really needed.